pep-0705.rst

```diff
@@ -0,0 +1,96 @@
+PEP: 705
+Title: Recording provenance of installed packages
+Author: Fridolin Pokorny <fridolin.pokorny at gmail.com>, Trishank Karthik Kuppusamy <karthik@trishank.com>
+Sponsor:
+PEP-Delegate:
+Discussions-To: https://discuss.python.org/t/pep-705-recording-provenance-of-installed-packages/23340
+Status: Draft
+Type: Process
+Content-Type: text/x-rst
+Created: 30-Jan-2002
+Post-History:
+
+
+Abstract
+========
+
+This PEP describes a way to record provenance of Python packages installed.
+The record is created by an installer and is available to users in a form of a
+JSON file ``direct_url.json`` in ``.dist-info`` directory. The PEP is an
+extension to :pep:`610` for cases when installed packages are comming from a
+package index.
+
+
+Motivation
+==========
+
+Installing a Python package involves downloading the package from a source and
+extracting its content to an appropriate place. After the installation process
+is done, information about the artifact used as well as its source is generally
+lost. Nevertheless, there are use cases for keeping records of artifacts used
+for installing packages and their provenance.
+
+Python wheels can be built with different compiler flags or supporting
+different wheel tags.  In both cases, users might get into a situation in which
+multiple wheels might be considered by installers (possibly from different
+package indexes) and immediately finding out which artifact was actually used
+during the installation might be helpful. This way, tools reporting software
+installed, such as tools reporting SBOM (Software Bill of Material), might give
+more accurate reports.
+
+The motivation described in this PEP is a direct extension to :pep:`610`.
+Besides stating information about packages installed from an URL, installers
+SHOULD record information also for packages installed from Python package
+indexes or from filesystem.
+
+
+Examples
+========
+
+An example of a ``direct_url.json`` file in the ``.dist-info`` directory
+alongside files stated in :pep:`376` and further adjusted in :pep:`627`. The
+specified file stores a JSON describing artifact used to install a package as
+well as its source.  The record is similar to ``direct_url.json`` described in
+:pep:`610`:
+
+.. code::
+
+  {
+    "archive_info": {
+      "hash": "sha256=714ac14496c3e68c99c29b00845f7a2b85f3bb6f1078fd9f72fd20f0570002b2"
+    },
+    "url": "https://files.pythonhosted.org/packages/ed/35/a31aed2993e398f6b09a790a181a7927eb14610ee8bbf02dc14d31677f1c/packaging-23.0-py3-none-any.whl"
+  }
+
+If a source distribution was used to build a wheel file which was subsequently
+installed, the ``url`` must state URL to the source distribution used.
```

Comment:
You should be consistent about capitalizing RFC-2119 language here in the normative spec (which I'm not clear where that even is, since there's no Specification section, just an example).

```suggestion
installed, the ``url`` MUST state URL to the source distribution used.
```